src/tutorials/0005-regular-files-api/04.md

```diff
@@ -3,28 +3,35 @@
+The `cat` method returns an Async Iterable, that iterates over the file's chunks of data, i.e. Buffers.
+
+A `Buffer` is just a raw collection of bytes and as such doesn't make any assumptions about encodings or the type of data it contains. However, if we know the file being retrieved is a plain text file such as a `.txt`, we can convert its buffered contents to a UTF-8 string (an interpretation of those raw bytes) by calling the JavaScript method `.toString()`.
 
 So if you had the CID for a text file in an IPFS node, you could retrieve the file's contents as a readable string like so:
 
 ```javascript
-let bufferedContents = await ipfs.cat('QmWCscor6qWPdx53zEQmZvQvuWQYxx1ARRCXwYVE4s9wzJ') // returns a Buffer
-let stringContents = bufferedContents.toString() // returns a string
+const concat = require('it-concat')
+
+const bufferedContents = await concat(ipfs.cat('QmWCscor6qWPdx53zEQmZvQvuWQYxx1ARRCXwYVE4s9wzJ')) // returns a Buffer
+const stringContents = bufferedContents.toString() // returns a string
 ```
-Notice how the example above waits for the results of the `cat` method before converting the value to a string. To accomplish that in a single line of code, you'd need to wrap the `await` statement in parentheses like so:
+
+Notice how the example above waits for the concatenation of the results of the `cat` method (using the [`it-concat`](https://www.npmjs.com/package/it-concat) package) before converting the value to a string. To accomplish that in a single line of code, you'd need to wrap the `await` statement in parentheses like so:
 
 ```javascript
-let stringContents = (await ipfs.cat(cid)).toString() // returns a string
+const stringContents = (await concat(ipfs.cat(cid))).toString() // returns a string
 ```
 
-When you're ready to try this in the real world, you should note that the `cat` method can result in heavy memory usage, depending on the contents of the file being read. If you find this to be the case, you might want to explore the [catReadableStream](https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#catreadablestream) or [catPullStream](https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#catpullstream) methods instead.
+When you're ready to try this in the real world, you should note that the above example can result in heavy memory usage, depending on the contents of the file being read. If you find this to be the case, you might want to skip the usage of the `it-concat` package and process each chunk of data iteratively.
+This is the main reason why ipfs returns `Async Iterables` so that we can decide if we want to be aware of memory performance issues or not.
+But in ProtoSchool tutorials, we don't really need to worry about this since our code challenges will use small files, so we can concatenate everything without worrying about performance.
```

Comment:
```suggestion
In ProtoSchool tutorials, we don't really need to worry about this since our code challenges use small files, so we can concatenate everything without worrying about performance.
```